File: workspace/pset6/Sentimental/caesar/caesar.py
from cs50 import get_string
import sys

def main():
    #print("Number of arguments:", len(sys.argv), "arguments.")
    #print("Argument List:", str(sys.argv))

    if(len(sys.argv) != 2):
        print("Error: Expected only ONE KEY parameter")

    key = int(str(sys.argv[1]))
    string = get_string("plaintext: ")
    msg = list(string)
    newMsg = modifymsg(msg, key)
    print("ciphertext: ", end = "")
    print(''.join(newMsg))



def modifymsg(msg, key):
    i = 0
    ciphertext = []
    val = 65
    while(i < len(msg)):
        if((msg[i] >= 'A' and msg[i] <= 'Z') or (msg[i] >= 'a' and msg[i] <= 'z')):
            if(isLower(msg[i]) == True):
                val = 97
            else:
                val = 65
            newChar = (ord(msg[i]) - val + key) % 26 + val
            ciphertext.append(chr(newChar))
        i = i + 1
    return ciphertext

def isLower(c):
    if(c >= 'a'and c <= 'z'):
        return True
    else:
        return False


if __name__ == "__main__":
    main()

File: workspace/pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

void modifymsg(char msg[], int k);
bool isLower(char c);

int main(int argc, string argv[]){
    if(argc != 2){
        printf("Error: Expected only 1 parameter.\n");
        return 1;
    }
    int i = 0;
    int k = atoi(argv[1]);
    string msg = get_string("plaintext: ");

    modifymsg(msg, k);
    printf("ciphertext: ");
    while(msg[i] != '\0'){
        printf("%c", msg[i]);
        i++;
    }
    printf("\n");
}

void modifymsg(char msg[], int k){
    int i = 0;
    while(msg[i] != '\0'){
        if((msg[i] >= 'A' && msg[i] <= 'Z') || (msg[i] >= 'a' && msg[i] <= 'z')){
            if(isLower(msg[i])){
                msg[i] = (msg[i] - 97 + k) % 26 + 97;
            } else{
                msg[i] = (msg[i] - 65 + k) % 26 + 65;
            }
        }
        i++;
    }
}

bool isLower(char c){
    if(c >= 'a' && c <= 'z'){
        return true;
    }
    return false;
}